apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ResponseFieldSpec.kt

```diff
@@ -20,7 +20,7 @@ class ResponseFieldSpec(
     val context: CodeGenerationContext
 ) {
   fun factoryCode(): CodeBlock {
-    val factoryMethod = FACTORY_METHODS[responseFieldType]!!
+    val factoryMethod = FACTORY_METHODS[responseFieldType] ?: error("Unrecognized responseFieldType: $responseFieldType")
```

Comment:
The error with `!!` is cryptic. This was an official warning. Replacing with error results into much better error message.